src/xmipp/libraries/data/ctf.cpp

```diff
@@ -414,19 +414,18 @@ void CTFDescription1D::readFromMdRow(const MDRow &row, bool disable_if_not_K)
 
 void CTFDescription1D::readFromMetadataRow(const MetaData &md, size_t id, bool disable_if_not_K)
 {
-    MDRow row;
-    md.getRow(row, id);
-    readFromMdRow(row, disable_if_not_K);
+    std::unique_ptr<const MDRow> row = std::move(md.getRow(id));
```

Comment:
TBH, I don't understand what this method does. The MD should not change, the the bool is passed by value ...
Are you sure that row won't be deleted from MD once this unique pointer goes out of scope?

Reply:
Actual row won't be deleted, row remains in metadata. `MdRow` object will be deleted, but that's not a problem as `readFromMdRow` method only uses row once and then it could be deleted.